mutation_partition.cc

```diff
@@ -2038,13 +2038,63 @@ reconcilable_result reconcilable_result_builder::consume_end_of_stream() {
                                std::move(_memory_accounter).done());
 }
 
+template<FlattenedConsumer Delegate>
+class position_recording_consumer {
+    Delegate _delegate;
+    std::optional<dht::decorated_key> _pkey;
+    std::optional<clustering_key> _ckey;
```

Comment:
The underlying consumer (compact_mutation_state) does not remember the clustering key as far as I can see.
It does record the partition key. 

I don't want to extend the consumer to record the key as well to avoid impacting uses which don't need it.